sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/objects.scala

```diff
@@ -207,7 +211,8 @@ object ObjectSerializerPruning extends Rule[LogicalPlan] {
     }
   }
 
-  def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+  def apply(plan: LogicalPlan): LogicalPlan = plan.transformWithPruning(
+    _.containsAllPatterns(OBJECT_CONSUMER, PROJECT), ruleId) {
```

Comment:
why do we need `OBJECT_CONSUMER` here?

Reply:
Done. Updated to use SERIALIZE_TO_OBJECT. 
(ObjectConsumer is a super class of SerializeToObject. I thought that OBJECT_CONSUMER could cover more subclasses of ObjectConsumer, but it turns out other subclasses are not used in transform calls.)

Reply:
@sigmod also, I am surprised that the tests passed....

Reply:
@gengliangwang  Using OBJECT_CONSUMER doesn't affect correctness of the rule, because ObjectConsumer is a super class of SerializeFromObject.

Reply:
I see. I didn't refresh the page and see your comment. Thank you.

Reply:
Thanks for reviewing it, @gengliangwang!